lua/toggleterm.lua

```diff
@@ -110,11 +110,24 @@ end
 function M.handle_term_enter()
   local _, term = terms.identify()
   if term then
-    term:resize()
+    local ui = require("toggleterm.ui")
+    local persist_size = require("toggleterm.config").get("persist_size")
+    if term:is_split() and not (persist_size and ui.persistent[term.direction]) then
+      ui.resize_split(term)
+    end
     close_last_window(term)
   end
 end
 
+function M.handle_term_leave()
+  local _, term = terms.identify()
+  if term then
+    if term:is_float() then
+      term:close()
+    end
+  end
```

Comment:
this can be combined I think to
```suggestion
if term and term:is_float() then
  term.close()
end
```